backend/common/src/main/java/ai/verta/modeldb/common/futures/DoublingBackoffStrategy.java

```diff
@@ -0,0 +1,28 @@
+package ai.verta.modeldb.common.futures;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiPredicate;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Value;
+
+@Value
+@Getter(AccessLevel.NONE)
+class DoublingBackoffStrategy<R> implements RetryStrategy<R> {
+  private static final int INITIAL_DELAY_MILLIS = 10;
+  BiPredicate<R, Throwable> resultChecker;
+  int maxRetries;
+  AtomicInteger numberRetried = new AtomicInteger();
+
+  @Override
+  public Retry shouldRetry(R result, Throwable throwable) {
+    boolean apply = resultChecker.test(result, throwable);
```

Comment:
I'm not in love with the name `apply`. Maybe something like `success`?

Reply:
yes. this was left over from a refactoring. will definitely clean up. good catch!